includes/class.llms.review.php

```diff
@@ -212,23 +194,81 @@ public static function output() {
 	 *
 	 * @since 1.2.7
 	 * @since 5.9.0 Stop using deprecated `FILTER_SANITIZE_STRING`.
+	 * @since [version] Now checking if the user can write a review.
 	 *
 	 * @return void
 	 */
 	public function process_review() {
+		// Check the nonce.
+		if ( ! isset( $_POST['llms_review_nonce'] ) || ! wp_verify_nonce( sanitize_key( $_POST['llms_review_nonce'] ), 'llms-review' ) ) {
+			return;
+		}
+
+		$parent_id = llms_filter_input_sanitize_string( INPUT_POST, 'pageID' );
+
+		// Make sure the curren user can write reviews yet.
+		if ( ! self::current_user_can_write_review( $parent_id ) ) {
+			return;
+		}
 
 		$post = array(
 			'post_content' => llms_filter_input_sanitize_string( INPUT_POST, 'review_text' ), // The full text of the post.
 			'post_name'    => llms_filter_input_sanitize_string( INPUT_POST, 'review_title' ), // The name (slug) for your post.
 			'post_title'   => llms_filter_input_sanitize_string( INPUT_POST, 'review_title' ), // The title of your post.
 			'post_status'  => 'publish',
 			'post_type'    => 'llms_review',
-			'post_parent'  => llms_filter_input_sanitize_string( INPUT_POST, 'pageID' ), // Sets the parent of the new post, if any. Default 0.
+			'post_parent'  => $parent_id, // Sets the parent of the new post, if any. Default 0.
```

Comment:
The comment says we're defaulting to 0, but in `current_user_can_write_review` we're fetching `get_the_ID()`? Though we're not using it, possibly because of a typo (ie. `$parent_id = get_the_ID()` instead of `$post_id = get_the_ID()`

Reply:
Always have a parent_id now. https://github.com/gocodebox/lifterlms/pull/2592/commits/88ccd31f58175b7f2e63d2da7ac58677d0c5160b